### DISCUSSION TITLE
When using automerge, bbUseDefaultReviewers doesn't have the same behavior as regular reviewers

### DISCUSSION THREAD

**Discussion by nonell-viavi** [renovatebot/renovate#26777] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovate:37.92-slim

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### What is your question?

Hi,

First of all, thank you for this amazing product, honestly. Now to the "issue":

In bitbucket, we have several repositories with **default reviewers** configured with a set of users.
To reduce noise generated by renovate, we are trying to enable automerging of minors and patch when tests pass, using this config:

```
"packageRules": [
    {
            "matchManagers": ["poetry"],
            "matchUpdateTypes": ["minor", "patch"],
            "automerge": true
    }
]
```

My expectation at this point, was that the reviewers would only be added to the pull request IF the tests failed, as the documentation for automerge [says](https://docs.renovatebot.com/key-concepts/automerge/#assignees-and-reviewers).

But what we see is that default Reviewers are **Always** added if the `bbUseDefaultReviewers==true`. And based of my very shallow understanding of [the code](https://github.com/renovatebot/renovate/blob/f634b9c9883d1dd8d73e8652a3d3ae4175d5aa88/lib/modules/platform/bitbucket-server/index.ts#L805), it seems to be intended behavior. Copying it here for easy reference:

```
  let reviewers: BbsRestUserRef[] = [];

  
  if (platformOptions?.bbUseDefaultReviewers) {
    ...
    reviewers = /* Get default reviewers for repo from bitbucket server*/
  }

  const body: PartialDeep<BbsRestPr> = {
    title,
    description,
    fromRef: {
      id: `refs/heads/${sourceBranch}`,
    },
    toRef: {
      id: `refs/heads/${base}`,
    },
    reviewers,            // <- Always adds those reviewers, no logic regarding to "automerge".
  };

  let prInfoRes: HttpResponse<BbsRestPr>;
  try {
    prInfoRes = await bitbucketServerHttp.postJson<BbsRestPr>(
      `./rest/api/1.0/projects/${config.projectKey}/repos/${config.repositorySlug}/pull-requests`,
      { body },  
    );
  ...
  
```
 
My question is: **Should the `bbUseDefaultReviewers` play nice with automerge?**

The other options that I see are:
- duplicate the default reviewers from bitbucket into the `reviewers` field in `renovate.json`. Not liking it because of the duplication.
- Use `automergeType=branch`, and allow renovate user to merge to all those repositories without a PR. It looks like an option, implies changing config in each repo.
- Move default reviewers to `CODEOWNERS`. I'm not too familiar with this option, I don't know yet how bitucket plays with `CODEOWNERS` although it seems supported.

Thank you!

**Comment by rarkins** [comment]
I'm not sure there's any need to assign reviewers in this case so maybe it can be changed to behave as you expect

**Reply by nonell-viavi** [reply]
Perfect! I am ok contributing with a Pull Request, if that is acceptable.

As I am not familiar with the code base, I would like to ask for a few pointers (I'll throw the question here, but happy to open the PR and continue the discussion over there).

My approach would probably be to add code to this block:
https://github.com/renovatebot/renovate/blob/23a334c7e34c0a9eba48697f6094dd52e615000b/lib/workers/repository/update/pr/participants.ts#L73-L82

With something among the lines of:

```
if ( config.bbUseDefaultReviewers ) {
  reviewers = reviewers.concat(platform.getDefaultReviewers()) // Change that to return an empty list for the other platforms?
}
```

I just don't like checking for bitbucket directly in that code, feels like it goes against the modularization that exists in the code.

**Reply by viceice** [reply]
no, leave it inside bitbucket platform code.

**Reply by nonell-viavi** [reply]
Hi,

Probably it is my lack of familiarity, but I am not seeing a way to add the desired functionality while limited to only modify bitbucket platform code [lib/modules/platform/bitbucket-server](https://github.com/renovatebot/renovate/tree/main/lib/modules/platform/bitbucket-server).

The function `addReviewers` takes already the reviewers list:
https://github.com/renovatebot/renovate/blob/0b0728791803a6dd371b7ca5706e04cad5790a57/lib/modules/platform/bitbucket-server/index.ts#L581-L584

So presumably, the code that calls `addReviewers` is what would need to fetch bitbucket's default reviewers, and I thought that code was in participants.ts.

Could you throw me some pointers in how you would like that implemented?

And thank you for your time on this
